src/transformers/training_args.py

```diff
@@ -470,10 +470,10 @@ class TrainingArguments:
     sharded_ddp: str = field(
         default="",
         metadata={
-            "choices": ["simple", "zero_dp_2", "zero_dp_3", "zero_dp_2 offload", "zero_dp_3 offload"],
```

Comment:
Choices started to grow longer as there are 4 more possibilities of options now. Can put it back if you really want to but I think the doc is enough.

Reply:
but why do all combinations? why not just list the components

Reply:
"choices": ["simple", "zero_dp_2", "zero_dp_3", "offload", "auto_wrap"],

Reply:
This instruction will fail if we pass anything not in the list of choices, like `"zero_dp_2 offload".

Reply:
oh, I understand! It doesn't know of composite values and would have failed anyway if one were to pass: "offload zero_dp_2 offload" (reverse).

So it's either subclassing this argparse behavior to support compose values or removing it as you did. 

I think as long as the help entry for that option lists all the choices, which it does, then it's good enough.